src/Services/Session.vala

```diff
@@ -109,26 +123,34 @@ public class Notifications.Session : GLib.Object {
     }
 
     public void clear () {
-        try {
-            FileUtils.set_contents (session_file.get_path (), "");
-        } catch (FileError e) {
-            warning (e.message);
-        }
-    }
-
-    private void create_session_file () {
-        try {
-            session_file.create (FileCreateFlags.REPLACE_DESTINATION);
-        } catch (Error e) {
-            warning (e.message);
-        }
+        key = new KeyFile ();
+        write_contents ();
     }
 
+    //This can get called multiple times in quick succesion. Wait until not called for 150 ms before writing file.
+    private uint write_contents_timeout_id = 0;
+    private bool can_write_contents = false;
     private void write_contents () {
-        try {
-            FileUtils.set_contents (session_file.get_path (), key.to_data ());
-        } catch (FileError e) {
-            warning (e.message);
+        if (write_contents_timeout_id > 0) {
+            can_write_contents = false;
```

Comment:
I feel it would be cleaner to simply remove the existing timeout, set the id to 0 and not return in this if statement, then there is no need for the extra boolean?

Reply:
I did not want to keep destroying and recreating the timeout every time a signal arrives within the timeout delay (which could be dozens of times if there are a lot of notifications).  Not sure what overhead that represents, however, and code clarity might override efficiency I guess.